lib/setup.py

```diff
@@ -48,7 +48,7 @@
     # pyarrow is not semantically versioned, gets new major versions frequently, and
     # doesn't tend to break the API on major version upgrades, so we don't put an
     # upper bound on it.
-    "pyarrow>=6.0",
+    "pyarrow>=7.0",
```

Comment:
The `nbytes` in pyarrow 6.0 has a bug. But I think it's fine to just pin this up to `7.x` since it is also already ~2 years old.